lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2569,18 +2569,75 @@ void AttributeChecker::visitPropertyDelegateAttr(PropertyDelegateAttr *attr) {
     nominal->diagnose(diag::property_delegate_reserved_name);
   }
 }
+
 // SWIFT_ENABLE_TENSORFLOW
 /// Returns true if the given type conforms to `Differentiable` in the given
 /// module.
-static bool conformsToDifferentiableInModule(Type type, ModuleDecl *module) {
-  auto &ctx = module->getASTContext();
+static bool conformsToDifferentiable(TypeChecker &TC, Type type,
+                                     DeclContext *DC) {
   auto *differentiableProto =
-      ctx.getProtocol(KnownProtocolKind::Differentiable);
-  return LookUpConformanceInModule(module)(
-      differentiableProto->getDeclaredInterfaceType()->getCanonicalType(),
-      type, differentiableProto).hasValue();
+      TC.Context.getProtocol(KnownProtocolKind::Differentiable);
+  return TC.conformsToProtocol(
+      type, differentiableProto, DC,
+      ConformanceCheckFlags::Used | ConformanceCheckFlags::InExpression)
+          .hasValue();
 };
 
+// SWIFT_ENABLE_TENSORFLOW
+/// Creates a `AutoDiffParameterIndices` for the given function type,
+/// inferring all differentiation parameters.
+/// The differentiation parameters are inferred to be:
+/// - All parameters of the function type that conform to `Differentiable`.
+/// - If the function type's result is a function type, then also all
+///   parameters of the function result type that conform to `Differentiable`.
+static AutoDiffParameterIndices *
+inferParameters(AnyFunctionType *functionType, TypeChecker &TC,
```

Comment:
The name here is a bit too general for `TypeCheckAttr.cpp`. How about `inferDifferentiableParameters`?